netmiko/cisco/cisco_xr.py

```diff
@@ -105,36 +114,74 @@ def commit(
         else:
             command_string = "commit"
 
+        if force:
+            command_string = command_string.replace("commit", "commit force")
+        if best_effort:
+            command_string = command_string.replace("commit", "commit best-effort")
+        if replace:
+            command_string = command_string.replace("commit", "commit replace")
+        print(f"command string is {command_string}")
+
         # Enter config mode (if necessary)
         output = self.config_mode()
 
-        # IOS-XR might do this:
-        # This could be a few minutes if your config is large. Confirm? [y/n][confirm]
-        new_data = self._send_command_str(
-            command_string,
-            expect_string=r"(#|onfirm)",
-            strip_prompt=False,
-            strip_command=False,
-            read_timeout=read_timeout,
-        )
-        if "onfirm" in new_data:
-            output += new_data
+        if replace:
+            new_data = self._send_command_str(
+                command_string,
+                expect_string=r"This commit will replace or remove the entire running configuration",
+                strip_prompt=False,
+                strip_command=False,
+                read_timeout=read_timeout,
+            )
+            if "This commit will replace or remove the entire running configuration" in new_data:
+                output += new_data
+                new_data = self._send_command_str(
+                    "yes",
+                    expect_string=r"#",
```

Comment:
this should be r'\)#$'

Reply:
This is the same code from the origin repo. https://github.com/ktbyers/netmiko/blob/426171102d259d2caebd9e6ac0b78f4ed3cb3758/netmiko/cisco/cisco_xr.py#L124

Reply:
Trying to keep minimal custom changes unless they are absolute necessary.